Microsoft.Azure.Cosmos/src/HttpClient/CosmosHttpClientCore.cs

```
@@ -96,20 +119,25 @@ internal sealed class CosmosHttpClientCore : CosmosHttpClient
 
         public static HttpMessageHandler CreateHttpClientHandler(int gatewayModeMaxConnectionLimit, IWebProxy webProxy)
         {
+            HttpClientHandler httpClientHandler = new HttpClientHandler();
+
+            // Proxy is only set by users and can cause not supported exception on some platforms
+            if (webProxy != null)
+            {
+                httpClientHandler.Proxy = webProxy;
```

WebProxy also throws PlatformNotSupported on WASM, if the user is creating a NETStandard library that they are then sharing on a Web App and WASM app, they could be setting the Proxy on both cases, but only one would apply. With this change, the Web App would apply it and the WASM would throw an exception. Before, both Apps would work, just that the WASM would not apply it due to the limitation on the platform.

---

It seems dangerous that we are secretly bypassing the proxy for WASM users. The better model is to throw the exception to the user so they know it's not handled rather than it just ignoring the setting the customer spending a lot of time trying to figure out why the proxy is not getting used.